sympy/physics/continuum_mechanics/beam.py

```diff
@@ -209,6 +210,68 @@ def bc_deflection(self):
     def bc_deflection(self, d_bcs):
         self._boundary_conditions['deflection'] = d_bcs
 
+    def apply_support(self, loc, type="cantilever"):
+        """
+        This method applies support to a particular beam object.
+
+        Parameters
+        ==========
+        loc : Sympifyable
+            Location of point at which support is applied.
+        type : String
+            Determines type of Beam support applied.
+            - For Simple pinned Support, type = "pin"
+            - For Cantilever Support, type = "fixed"
+            - For Rolling Support, type = "roller"
+
+        Examples
+        ========
+        There is a beam of length 30 meters. A moment of magnitude 120 Nm is
+        applied in the clockwise direction at the end of the beam. A pointload
+        of magnitude 8 N is applied from the top of the beam at the starting
+        point. There are two simple supports below the beam. One at the end
+        and another one at a distance of 10 meters from the start. The
+        deflection is restricted at both the supports.
+
+        Using the sign convention of upward forces and clockwise moment
+        being positive.
+
+        >>> from sympy.physics.continuum_mechanics.beam import Beam
+        >>> from sympy import symbols
+        >>> E, I = symbols('E, I')
+        >>> b = Beam(30, E, I)
+        >>> b.apply_load(-8, 0, -1)
+        >>> b.apply_support(10, 'roller')
+        >>> b.apply_support(30, 'roller')
```

Comment:
What is the difference in applying a supprt and a load? Didn't we add supports by applying unknown loads before? Is this method really necessary?

Reply:
I guess it does let you track more carefully what the unknown loads are. So maybe worth it. Do you use that info internally?

Reply:
Yeah, we can add support before applying load too. But I didn't get you by which info is to be used internally?